### ISSUE TITLE
Add queryjump to redirect url

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#3206]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

Mentioned in https://github.com/uBlockOrigin/uBlock-issues/issues/760#issuecomment-720079511 https://github.com/uBlockOrigin/uBlock-issues/issues/2476#issuecomment-1443552457 and now it should be possible to implement as a trusted network filter modifier. It's similar to `$uritransform` except for this should redirect to different domain. Maybe `$to` or something can be combined to restrict the domain to redirect rather than always blindly trust the url in parameter. Though the original context was about shortners, I rather want to use this to automate skip of strict-block page which is currently possible only by manual click.

### A specific URL where the issue occurs.

```yaml
`https://ouo.io/s/BulJXu78?s=http://hackstore.link/alg2i`
```

### Steps to Reproduce

NA

### Expected behavior

NA

### Actual behavior

NA

### uBO version

1.57.0

### Browser name and version

Not relevant

### Operating System and version

Windows 10

**Comment by garry-ut99** [comment]
> Mentioned in (...)

Here too: 
- https://github.com/uBlockOrigin/uBlock-issues/issues/2693, which redirects to:
- https://github.com/uBlockOrigin/uBlock-issues/issues/1784

it was declined in the past, but yes, it then had been mentioned to be reconsidered over time as per : https://github.com/uBlockOrigin/uBlock-issues/issues/1784#issuecomment-991645988

**Comment by gorhill** [comment]
Test cases:

---

Filters: 

    ||example.com^$doc
    ||example.com/path/to/tracker$urlskip=?url

Navigate to `https://example.com/path/to/tracker?url=https://example.org/`

Navigated to: `https://example.org/`

---

Filter:

    ||example.com/path/to/tracker$urlskip=?url ?to

Navigate to `https://example.com/path/to/tracker?url=https%3A%2F%2Fexample.org%2Fpath%2Fto%2Ftracker%3Fto%3Dhttps%253A%252F%252Fgithub.com%252F`

Navigated to: `https://github.com/`

---

Filter: 

    ||example.com/path/to/tracker$urlskip=?url ?to

Navigate to `https://example.com/path/to/tracker?url=https%3A%2F%2Fexample.org%2Fpath%2Fto%2Ftracker`

Navigated to `https://example.com/path/to/tracker?url=https%3A%2F%2Fexample.org%2Fpath%2Fto%2Ftracker`

Note: No skip occurred since not all extraction directives could be fulfilled

**Comment by krystian3w** [comment]
Spaces before "?" are needed?

**Comment by stephenhawk8054** [comment]
I think the parameter value should be a valid string to use with `new URL()`. The string `hepsiburada.com/philips-ca6700-10-espresso-makinesi-kirec-temizleme-solusyonu-p-HBCV00002YNJSF` is not a valid string for URL, i.e

```js
new URL("hepsiburada.com/philips-ca6700-10-espresso-machine-kirec-temizleme-solusyonu-p-HBCV00002YNJSF")
```

will return error, so no, you can't use `urlskip` in this case.

**Comment by gorhill** [comment]
> `||rdr.btrck.com/aff_c*$urlskip=?offer_id aff_id source aff_sub aff_sub2 aff_sub3`

That is not a valid `urlskip` filter, it doesn't match what the documentation says how it works. Looks like you want to extract the URL from the `aff_sub31` parameter; `offer_id`, `aff_id`, `source`, `aff_sub`, `aff_sub2` are not valid extraction directives. This would be the proper filter:

    ||rdr.btrck.com/aff_c*$urlskip=?aff_sub3

However as @stephenhawk8054 pointed out, the extracted string is not a valid URL as per URL API, so it does not work in the end.

I do think it should work though, so I will see what can be done about it.

**Comment by gorhill** [comment]
In the next dev build, this should work:

    ||rdr.btrck.com/aff_c?$urlskip=?aff_sub3 +https

**Comment by garry-ut99** [comment]
Works fine for me (Firefox & Chromium).

**Comment by Catman-232** [comment]
Sorry to bump this but I really didn't think this was worth making a whole new issue for, I know this was added in the past couple weeks so I haven't been able to find any documentation or anyone asking about this case scenario: How do I use $urlskip if the hyperlink URL doesn't even have a property after the `?` ? From what I can tell the URL API then just considers the whole URL as a property instead of a value. 

For convenience I was using this to successfully skip Pixiv's kinda reundant outbound link capture page `https://www.pixiv.net/jump.php` since on user profiles they have social media icons for Twitter etc which can only actually be customised with an account name, not any potentially malicious URL. Those links use `url=`, so that's not a problem to catch with a rule.

However, on gallery pages the links in the description actually do not have `url=`, they are just `https://www.pixiv.net/jump.php?https%3A%2F%2Fx.com%2FExample_Username`. Obviously I can't make a rule like `||pixiv.net/jump.php$urlskip=?` since that's "Invalid filter: Bad value assigned to a valid option. I was using uBO for this since I figured it would be more direct and  performant than making some Tampermonkey script to do any kind of instant redirect or href edits on all the hyperlinks, convenient coincidence that this static network modifier filter was recently added and can do what I was hoping for, as long as the URL has a property to target.

**Comment by gorhill** [comment]
> How do I use $urlskip if the hyperlink URL doesn't even have a property after the `?`

The initial version of `urlskip=` is limited and deal with the obvious `?url=` way.

People can then report cases which cannot be handled by the current `urlskip=`, then we will consider adding support. For example, see <https://github.com/gorhill/uBlock/commit/e81e70937f>.

So what you are reporting here is a case where the URL to skip to is encoded as a parameter name. I will think if and what could be done about this.

**Comment by Catman-232** [comment]
Wow that was a fast response, thanks. Sorry about the example link, I didn't realise it wouldn't work, I see that if the link is put straight into the address bar it just takes you to an error page. The a tags in the HTML use `href="/jump.php?https...etc` which I guess means it only works in-site. I've found an example for you that I confirmed worked while not logged in (using a private window) https://www.pixiv.net/en/artworks/122985712
I didn't even know they had an advert on that page hah.

I was starting a bit of a ramble about the properties in URL objects but after seeing your post edit I figured you probably would already know about this anyway:
<details><summary>the start of the ramble</summary>So I was still tinkering with this and tried doing `new URL` in the dev tools console like stephenhawk8054 showed, and the URL object has the destination URL as a `search` property containing the encoded chars and the question mark at the beginning. It also has a `searchParams` property containing a `URLSearchParams` object with one entry ...</details>

**Comment by gorhill** [comment]
> `https://www.pixiv.net/en/artworks/122985712`

Where do I click on that page to trigger the navigation to the cloaked URL?

**Comment by Catman-232** [comment]
I suppose I could do with a lesson in being directly specific, sorry. It's these links down below the image, you can click "Continue Reading" to expand it to show the full description which has a lot of hyperlinks in it but most are just Fanbox and Patreon links. Actually looks like all of them except the `mangamaterials.fanbox.cc` is prepended with `https://www.pixiv.net/jump.php?`
![rightthere](https://github.com/user-attachments/assets/49ff2e56-e988-4e58-95ef-073196aecae6)

> > https%3A%2F%2Fx.com%2FExample_Username
> 
> That's percent encoding. Support for it will have to be added and doesn't exist yet.
It was what it said in the address bar on the in-between page so I just copied that, the socials links on profiles, which use `?url=` also show the encoding in the address bar and the $urlskip rule worked fine on them.

**Comment by gorhill** [comment]
This should work with 1.60.1b6 and above:

    ||pixiv.net/jump.php?$urlskip=&1

**Comment by Catman-232** [comment]
It works, I just missed the `?` moving to before `$urlskip=` the first time I tested it and had to look a little closer to see that.

**Comment by sertonix** [comment]
Is it possible to use use urlskip when the link is part of the path?
For example converting `https://www.podtrac.com/pts/redirect.mp3/dovetail.prxu.org/7057/1b6a25b9-d8fe-48cd-aab1-cec94c3ae137/darknet-diaries-ep150-mobman-2.mp3` into `https://dovetail.prxu.org/7057/1b6a25b9-d8fe-48cd-aab1-cec94c3ae137/darknet-diaries-ep150-mobman-2.mp3`

**Comment by gorhill** [comment]
> Is it possible to use use urlskip when the link is part of the path?

Yes, I expected such case would be reported eventually. On which website did you encounter such link?

**Comment by sertonix** [comment]
> On which website did you encounter such link?

https://darknetdiaries.com/episode/150/

When you click on the download button (cloud icon) and then click on mp3.

![image](https://github.com/user-attachments/assets/dabdd89b-205f-4da3-9fa2-6ea52c2426ff)

**Comment by gorhill** [comment]
This should work in the dev build 1.60.1b11 and above:

    ||podtrac.com/pts/redirect.mp3/$urlskip=/podtrac\.com\/pts\/redirect\.mp3\/(.*?\.mp3\b)/ +https

**Comment by D4niloMR** [comment]
We can use this for Reddit click tracking in emails, example link:

```
https://click.redditmail.com/CL0/https://www.reddit.com/r/natureisfuckinglit/comments/1f7l41z/orca_pod_saying_hi_to_paddle_boarder/%3F%2524deep_link=true&correlation_id=21bc5fa3-f0f4-43b5-92e4-6d2deb135b8a&post_fullname=t3_1f7l41z&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_author/1/01000191bd393638-90cd632e-f420-45fd-a732-27dc3e441ee2-000000/aBs5XoGsoLNHwf95BGM6JzVs4iNSldLNZ9hOBaWN_HA=369%5D
```

The only problem is that sometimes it's encoded like in the example link. I got this link from `https://urlscan.io/search/#click.redditmail.com`.

**Comment by gorhill** [comment]
> Reddit click tracking in emails

This works?

    ||click.redditmail.com/CL0/$urlskip=/CL0\/(https:\/\/.*?)%3F/

Is it always `/CL0/`?

**Comment by D4niloMR** [comment]
Yes, this works for the example link but won't work for the ones that are not encoded at all, or have the slash after `https` also encoded:

```
https://click.redditmail.com/CL0/https://www.reddit.com/r/natureisfuckinglit/comments/1f7l41z/orca_pod_saying_hi_to_paddle_boarder/?%24deep_link=true&correlation_id=21bc5fa3-f0f4-43b5-92e4-6d2deb135b8a&post_fullname=t3_1f7l41z&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_author/1/01000191bd393638-90cd632e-f420-45fd-a732-27dc3e441ee2-000000/aBs5XoGsoLNHwf95BGM6JzVs4iNSldLNZ9hOBaWN_HA=369
https://click.redditmail.com/CL0/https:%2F%2Fwww.reddit.com/r/natureisfuckinglit/comments/1f7l41z/orca_pod_saying_hi_to_paddle_boarder/%3F%2524deep_link=true&correlation_id=21bc5fa3-f0f4-43b5-92e4-6d2deb135b8a&post_fullname=t3_1f7l41z&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_author/1/01000191bd393638-90cd632e-f420-45fd-a732-27dc3e441ee2-000000/aBs5XoGsoLNHwf95BGM6JzVs4iNSldLNZ9hOBaWN_HA=369%5D
```

And the second link here can't be used in `urlskip` without `+https`

> Is it always `/CL0/`?

I have only seen it starting with `/CL0/`.

**Comment by stephenhawk8054** [comment]
Hmm...

```adb
||click.redditmail.com/CL0/$urlskip=/CL0\/(https?:(?:\/\/|%2F%2F)[^%]+)/
```

works for the first 2 links, but not the 3rd link although it's matched on regex101

**Comment by garry-ut99** [comment]
For some unknown reason I can't reply anymore in some commits, like [THIS ONE](https://github.com/uBlockOrigin/uAssets/commit/c7699d6469d9ec6dabc8602bb9783da5ac73858c#comments), even on a clean Firefox profile, because comments loading stucks infinitely: [ANIMATION](https://github.com/user-attachments/assets/ec68d350-d367-4b9b-ae48-43afd5f9efd7), hence I'm forced to put here my reply here, I'll bring a bit of the context here: 

On `https://hozana.si/` there is a play button to a song of which the URL contains 2 trackers: `https://chtbl.com/track/34D156/dts.podtrac.com/redirect.mp3/hozana.si/mp3/berila/467.mp3`

> gorhill: This should work but: `||chtbl.com/track/$urlskip=/\/track\/\w+\/([^?]+)/ +https`

As for the provided filter: the regex pattern correctly matches the URL, but the filter doesn't work, nor appears in the logger (tried on both Firefox & Chromium).

> gorhill: I would like to see more instances of that tracker to see whether the pattern holds

https://publicwww.com/websites/%22chtbl.com%2Ftrack%2F%22/
https://www.google.com/search?q=%22%22chtbl.com%2Ftrack%2F%22%22

Some URLs contain only  1 tracker, but  the pattern still holds:
`https://chtbl.com/track/9EDG43/storage.tuba.fm/podcast/7c1db3c0796d9ebd32128d6f7a9c3336/672c0300/00/00/16/49/86.mp3?v987`

Another URL requires another pattern/filter because it contains 4 trackers?: `https://pdst.fm/e/chtbl.com/track/43774/mgln.ai/e/143/pdrl.fm/2d1ab9/traffic.megaphone.fm/SONY7102299968.mp3?updated=1726235869` on (`https://www.podchaser.com/podcasts/infamous-5094597/episodes/a-lost-friend-the-matthew-perr-223437604`) 

One thing I noticed is that the provided patterns skip parameters behind `.mp3` for example: (if such parameters are considered tracking params / unwanted, and thus are skipped intentionally, then fine (just wanted to be sure):
-  a) `?v987` at the end of `https://chtbl.com/track/9EDG43/storage.tuba.fm/podcast/7c1db3c0796d9ebd32128d6f7a9c3336/672c0300/00/00/16/49/86.mp3?v987` (on `https://hozana.si/`)
- b) `?_=3` at the end of `https://chtbl.com/track/G3B499/podcasts.captivate.fm/media/e4549b58-0023-420d-a99e-bf690460130b/EP4-2010-2010-converted.mp3?_=3` (on `https://www.faxinapodcast.com/`)
- c) `?updated=1726235869` at the end of `https://pdst.fm/e/chtbl.com/track/43774/mgln.ai/e/143/pdrl.fm/2d1ab9/traffic.megaphone.fm/SONY7102299968.mp3?updated=1726235869` on (`https://www.podchaser.com/podcasts/infamous-5094597/episodes/a-lost-friend-the-matthew-perr-223437604`)

**Comment by stephenhawk8054** [comment]
Yeah, for redirection chain, it's better to write filter for each tracking domain

```adb
||pdst.fm/e/$urlskip=/\/e\/([^?]+)/ +https
||chtbl.com/track/$urlskip=/\/track\/\w+\/([^?]+)/ +https
||mgln.ai/e/$urlskip=/\/e\/\w+\/([^?]+)/ +https
||pdrl.fm^$urlskip=/pdrl\.fm\/\w+\/([^?]+)/ +https
```

**Comment by garry-ut99** [comment]
Yes, it works perfectly: [SCREENSHOOT](https://github.com/user-attachments/assets/5c9ed7fa-f158-44ea-8e1a-b4c5e4f1efcb)

**Comment by garry-ut99** [comment]
> On `https://hozana.si/` there is a play button to a song of which the URL contains 2 trackers: `https://chtbl.com/track/34D156/dts.podtrac.com/redirect.mp3/hozana.si/mp3/berila/467.mp3`

> ```adb
> ||podtrac.com^*redirect.mp3/$urlskip=/\/redirect\.mp3\/(.*?\.mp3\b)/ +https
> ||chtbl.com/track/$urlskip=/\/track\/\w+\/([^?]+)/ +https
> ```

Works when the URL is pasted into browser bar and has `doc` type in the logger: [SCREENSHOOT](https://github.com/user-attachments/assets/c477d13f-9e59-4f9c-914c-ad7d68745a10)

But doesn't work when visited `https://hozana.si/` and then: pressed "Play" button (`media` type in the logger) or from context menu > "Save sound as (file)" (`other` type in the logger) - [SCREENSHOOT](https://github.com/user-attachments/assets/c1f99c75-1b37-4d3b-980a-50647426e0ae).

It seems that `urlskip` works only for `doc` type (quoted from Wiki):

> ... is enforced only on top documents

**Comment by gorhill** [comment]
<https://github.com/gorhill/uBlock/commit/01eebffc1fd133f4f72f886551e79ba4bfc3bb24>

I added `-uricomponent`, so for these two links:

    https://click.redditmail.com/CL0/https://www.reddit.com/r/natureisfuckinglit/comments/1f7l41z/orca_pod_saying_hi_to_paddle_boarder/?%24deep_link=true&correlation_id=21bc5fa3-f0f4-43b5-92e4-6d2deb135b8a&post_fullname=t3_1f7l41z&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_author/1/01000191bd393638-90cd632e-f420-45fd-a732-27dc3e441ee2-000000/aBs5XoGsoLNHwf95BGM6JzVs4iNSldLNZ9hOBaWN_HA=369
    https://click.redditmail.com/CL0/https:%2F%2Fwww.reddit.com/r/natureisfuckinglit/comments/1f7l41z/orca_pod_saying_hi_to_paddle_boarder/%3F%2524deep_link=true&correlation_id=21bc5fa3-f0f4-43b5-92e4-6d2deb135b8a&post_fullname=t3_1f7l41z&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_author/1/01000191bd393638-90cd632e-f420-45fd-a732-27dc3e441ee2-000000/aBs5XoGsoLNHwf95BGM6JzVs4iNSldLNZ9hOBaWN_HA=369%5D

This works on my side:

    ||click.redditmail.com/CL0/$urlskip=/CL0\/.*?(www\.reddit\.com\/.+?)(?:\?|%3F)/ -uricomponent +https

**Comment by D4niloMR** [comment]
I looked some links on my email and I just need to remove the `\/` part and it works fine.

**Comment by gorhill** [comment]
Which `\/` did you need to remove? I don't understand why `\/` would prevent it from working.

**Comment by D4niloMR** [comment]
I removed the `\/` in the capture group because it won't match when it's `%2F` instead of `/`

So it becomes this:
```
||click.redditmail.com/CL0/$urlskip=/CL0\/.*?(www\.reddit\.com.+?)(?:\?|%3F)/ -uricomponent +https
```

**Comment by gorhill** [comment]
> It seems that `urlskip` works only for `doc` type

I can add support so urlskip images, media and beacon-like resources.

**Comment by garry-ut99** [comment]
Given the following two actions (on `https://hozana.si/`):
- 1] Pressing play button
- 2] Saving sound as file (from context menu)

On Chromium works fine for the both above actions, because the both actions are of `media` type in the logger [SCREENSHOOT](https://github.com/user-attachments/assets/70ca20e7-a3ac-4dcf-bb95-6274acf7f737).

However, on Firefox works only for action 1] which is of `media` type, and not for action 2] because it's of type `other`: [SCREENSHOOT](https://github.com/user-attachments/assets/a519173b-9bd1-4820-977c-1517241ac2a4) 

(however I'm not sure if the commit was even intended to support `other` type as well).

**Comment by krystian3w** [comment]
I don't know if this is expected behavior, but encoded tracking parameters are not removed from e.g. Google:

```adb
||doubleclick.net/aclk*adurl=$doc,_,urlskip=?adurl +https,_
||googleadservices.com/pagead/aclk*&adurl=$doc,_,urlskip=?adurl +https,_

```

    https://www.googleadservices.com/pagead/aclk?sa=L&ai=CrntXcI4SZ7XNE5Wyx_APlozr-AeOteTKernFxO3jErCQHxABINKXkzkoAmDplNCF8BqgAcSruJ8-yAEFqQLsP9tWJuqQPqgDAcgDApgECKoEnAJP0Cv_PqyRCuRYRCbHCNhegA7nHZaAs4NGeRa0jZtxGOZ1oibn2bDnEYQCC4gDDmy0_D-wF485zMwZK1_Ojwra-sLKC-cydk23iOCpGT4Nfm53oq0HRjUq-ATwLdYu20pEk5lovR46D0c_FWodyWvM_CHM9yz9iV25r1NFtiaZX--qKarX5TQlVeTf5dL2IEweioBCBQ47bZwsR79zHYfndAXslurWjdK8dXfZU_H2zncxkro_2q5_hOl1fwtnIaNNi0V-jIWcMZfUx_dWPcXb2V7HJHfeh2MU2LhzMfLmxqEkMCv0p1Fd3_q33dRRZQtwI0ThAq41QvdTPwflb5Z4OMEJh8qSk5eF4pSzZ1KI0zfQAKXRkk9UCszr3MAEn8qYu4AFiAWL0a2IUcAFiQGgBlPYBgKAB8TjiP8YmAcBqAfVyRuoB9m2sQKoB6a-G6gHjs4bqAeT2BuoB_DgG6gH7paxAqgH_p6xAqgHr76xAqgHusOxAqgHmgaoB_PRG6gHltgbqAeqm7ECqAeDrbECqAfgvbECqAf_nrECqAffn7ECqAe2rbECqAe0u7ECqAehqrECqAfetbECqAekr7ECqAf4wrECqAf7wrEC2AcB8gcCQGmgCMHePbAIDtIILgiR4YBQEAEYXzIHq4uA4L-AAToMgIABgICAgICUruADSL39wTpYgKzEr66YiQOxCeCFM8ikaVF7gAoUmAsByAsB0AsNogwIKgYKBOjCsQLaDBUKChDgs9DYp72YunMSAQUyBAoCMTSaDQENqg0CUEzIDQHYExOCFCkaJ21vYmlsZWFwcDo6Mi1jb20uc3VwZXJib3guYW9zLnBla29ibGFzdIgUA9AVAfgWAYAXAbIXAhgBuhcCOAGyGAkSAvBqGFMiAQDQGAE&ae=2&ase=3&asrd=1&gclid=EAIaIQobChMItcXFr66YiQMVFdkRCB0Wxhp_EAEYASAAEgKlXfD_BwE&num=1&sig=AOD64_0qlo2a384caTxnp5GyDclrJVw6ng&client=ca-app-pub-9704632099019185&fbs_aeid=8326360604868030464&rf=1&nb=8&nx=124&ny=656&mb=1&uach=WyJBbmRyb2lkIiwiIiwiIiwiIiwiIixudWxsLDEsbnVsbCwiIixudWxsLDBd&ms=CogEmsA_ATEaPgF_uwiQc5492-QA2u49tBMUjTeR1bWkof2-ErBM9277qgqcGpcnkfu8CiP1xVhAMAC_IBQOHMqL5X6r8Qu1EsADGKdgGiNYRwfqgw4-FA5UAhGJRzCOnyfoqq3178LnnetkKPxDa90wIHOZi0rNJ8XPi1_S4ycjREyA3DXTDgW_6KKKNVDMc4cnF-F5GiCzfjuIwS4QA4Z3nJQZvaKecGglEUldvxi6Y4giBbt6r8zHr-_M74_UVpTBX5vnJtZNp_40nFYNk5aUR5TY0VTRe_qzLomWRgn-ELjht-0QBpvWASRmg7mzqn9W6hTbp4voxEQyPXtElOQIBaTwFhVGuoEhdWg2lQTm2Ktbzps9Nj8-2sV4YqJ-p_StMS-U8hdpUzXtdDsx3J7JtF4X1CLOm4lvV0ade-9AxnaRYFlN0UcPEtWEQv3xQKYBSvOObkGoL_FMfs0LZDyq-jyndFNcvfaZgijbpehi-OAHtfNHUTXydbklbD2asnQCQ7SBKNRGlhJaZi957BnZh7QpZiVY518fUNC_PHsyJRGAJAulkjN9M-MB0MTHRMBW0MA7MR0lcKkSrOqHMUNs79I3EXRSkatLPwpZ42N-ssoadM6A-qi13OVJQS1OfrF1CXQZkCBTSm9NzCPJ60TbNP3br0_ON2f9sAZuksApJaLSxgd1eboW3CAE&dct=1&suid=25202308119&adurl=https://loteriavileda.pl/%3Futm_source%3Dyoutube%26utm_medium%3Dcpm%26utm_campaign%3Dloteria%26utm_content%3DBumper%26gad_source%3D5&label=video_click_to_advertiser_site

Goes to:

 `https://loteriavileda.pl/?utm_source=youtube&utm_medium=cpm&utm_campaign=loteria&utm_content=Bumper&gad_source=5`

 and refresh is needed to remove utm_ and gad_.

**Comment by garry-ut99** [comment]
I've checked on both Firefox and Chromium and I can't reproduce your issue with `AdGuard URL Tracking Protection` or `Actually Legitimate URL Shortener Tool` enabled, all params are removed properly and only `https://loteriavileda.pl/` is visited. 

Without above lists, on default config, params are not removed, but refreshing doesn't remove them either, so your issue still not reproduced, but it's expected because there are no `utm_` or `gad_` `removeparam` filters in uBlock filters, so I don't know why do you expect params to be removed. 

You need to enable an additional filter list, but if you have it already enabled, then I don't know why it doesn't work for you at the first visit, and you if you don't have it enabled, then I don't know how params for you are removed after refreshing. You didn't say whether you use default filter lists or have some additional filter lists enabled, so hard to guess what's going on. Maybe it's some issue on your side.

**Comment by stephenhawk8054** [comment]
Those params is encoded within `adurl` itself:

```
https://loteriavileda.pl/%3Futm_source%3Dyoutube%26utm_medium%3Dcpm%26utm_campaign%3Dloteria%26utm_content%3DBumper%26gad_source%3D5
```

so when it's skipped, it's skipped to the whole URL (including `utm_`...), and yeah, you'll need `removeparam` filters to remove those at the destination link.

**Comment by garry-ut99** [comment]
With `-uricomponent` no difference either.

**Comment by stephenhawk8054** [comment]
I'm testing this filter:

```adb
||com/*/http$doc,to=streak-link.com|streaklinks.com,urlskip=/\bcom\/[^\/]+\/https?%3A%2F%2F([^\/]+)/ +https -uricomponent
```

for these 2 links

```yml
https://streaklinks.com/BjW_1iyu7uvFNNxZXAei2V8L/http%3A%2F%2Fmy.hackmit.org
https://streak-link.com/Bomokso9PQsx_mCBXAc6oXuD/http%3A%2F%2Flaw.stanford.edu%2Fresearch%2Fsls-fellowships%2Fempirical-research-fellowship%2F
```

but not successful although `regex101.com` shows correct pattern. Anyone knows why?

**Comment by garry-ut99** [comment]
It's because this part: `||com/*/http$doc` alone doesn't match anything (page isn't strict blocked), hence the whole filter doesn't work. It works after changing it a bit, for example removing the  second `/`: `||com/*http$doc`. 

By the way none of the following work: 
```adb
||com/*/http$doc
||com^*/http$doc
||com/*^http$doc
||com^*^http$doc
```

All the following work:
```adb
||com/*http$doc (removed the second `/`)
||com/*/*http$doc (added another `*`)
||com/*http*$doc (added another `*`) 
(but two `*` in the same filter should be avoided for performance reasons, 
so these are just examples rather than to use it)
```

I didn't investigate deeper why your original filter didn't work, most likely some uBO parser quirk I guess (eventually a bug).

**Comment by stephenhawk8054** [comment]
Hmm... I see. You're right, both

```adb
||streaklinks.com/*/http$doc
||com/*http$doc
```

work, but

```adb
||com/*/http$doc
```

doesn't work.

@gorhill Sorry, do you know why?

**Comment by garry-ut99** [comment]
By the way, strange:
`||com/*/` - doesn't work
`||com/*` - works
`||streaklinks.com/*/` - doesn't work
`||streaklinks.com/*/http` - doesn't work
`||streaklinks.com/*/http$doc` - works

**Comment by stephenhawk8054** [comment]
For pattern-based filters

```adb
||com/*/
||streaklinks.com/*/
||streaklinks.com/*/http
```

they need `doc` to strict-block it. It blocks correctly if I add `doc` But looks like `||com/` doesn't need it, maybe uBO understands it as `||com^`?

**Comment by garry-ut99** [comment]
> For pattern-based filters

I guess you mean a wildcard-based pattern. Then looking at the logger and at the strict-blocking page, it shows `||com/*` as `||com/` hence `*` has been stripped out as unnessecary, and it's no longer a wildcard-pattern based filter hence doesn't need `$doc` anymore to strict-block.

**Comment by garry-ut99** [comment]
Getting back to the original issue:

> `||com/*/http$doc` doesn't work.

not a full answer, but I've noticed that it works with URL-decoded links:

```
https://streaklinks.com/BjW_1iyu7uvFNNxZXAei2V8L/http://my.hackmit.org
https://streak-link.com/Bomokso9PQsx_mCBXAc6oXuD/http://law.stanford.edu/research/sls-fellowships/empirical-research-fellowship/
```

it also works with original but truncated links:
```
https://streaklinks.com/BjW_1iyu7uvFNNxZXAei2V8L/http
https://streak-link.com/Bomokso9PQsx_mCBXAc6oXuD/http
```

strangely, it also works after replacing  `http` with `https` in the filter, so we get a new one:  `||com/*/https$doc` and then after replacing `http` with `https` in the original URL-encoded target links:
```
https://streaklinks.com/BjW_1iyu7uvFNNxZXAei2V8L/https%3A%2F%2Fmy.hackmit.org
https://streak-link.com/Bomokso9PQsx_mCBXAc6oXuD/https%3A%2F%2Flaw.stanford.edu%2Fresearch%2Fsls-fellowships%2Fempirical-research-fellowship%2F
```

**Comment by gorhill** [comment]
> Sorry, do you know why?

Yes. In the URL `https://streaklinks.com/BjW_1iyu7uvFNNxZXAei2V8L/http%3A%2F%2Fmy.hackmit.org`, there are no `http` token, there is `http%3A%2F%2Fmy`, which can be used as a token, though only the first 7 characters will be used, so `http%3A%`.

So for a filter like `||com/*/http$doc`, the extracted token can be either `com` or `http` (both are considered "bad" token by the way given there occurrences). The `http` one won't match the URL one, which is `http%3A%`. The way to solve this is by telling uBO that the `http` part is just the start of a token, like so `||com/*/http*$doc`.

**Comment by stephenhawk8054** [comment]
I see, so if a word is bound between a non-word character (`/`) and the type sign (`$`), that word will be tokenized?

**Comment by gorhill** [comment]
The dollar sign `$` is not a factor here, the extracted pattern to match was `||com/*/http`; uBO always assumes word boundary, including at the end of the pattern.

**Comment by stephenhawk8054** [comment]
Hmm... But how does `||streaklinks.com/*/http$doc` work? It has the same `http` token which also should not match the link?

**Comment by gorhill** [comment]
`||streaklinks.com/*/http$doc` works because the extracted (good) token is `streaklinks`.

**Comment by stephenhawk8054** [comment]
I see. Actually now I prefer splitting the filter to 2 filters with better tokens. It's not a case of multiple random domains anyway.

**Comment by gorhill** [comment]
I am looking at the two filters currently in _privacy.txt_:

    ||streaklinks.com/*/http$doc,urlskip=/\bstreaklinks\.com\/[^\/]+\/https?%3A%2F%2F([^\/]+)/ +https -uricomponent
    ||streak-link.com/*/http$doc,urlskip=/\bstreak-link\.com\/[^\/]+\/https?%3A%2F%2F([^\/]+)/ +https -uricomponent

Beside reversing the order of `+https -uricomponent`, I don't think there is a need to match the `\bstreaklinks\.com\/[^\/]+` in the regex given that the pattern was already a match. This could simply be:

    ||streaklinks.com/*/http$doc,urlskip=/\/https?%3A%2F%2F([^\/]+)/ -uricomponent +https
    ||streak-link.com/*/http$doc,urlskip=/\/https?%3A%2F%2F([^\/]+)/ -uricomponent +https

**Comment by gorhill** [comment]
Question regarding the 31 `urlskip=` filters already in _privacy.txt_: Is the purpose any one of these filters to redirect blocked network requests?

My question is because I am re-considering applying `urlskip=` to blocked network requests by default, I am thinking now that in case of blocked network requests, the redirection should be made explicit.

Apply the `urlskip=` redirect only if the network request is not blocked:
`||example.com/path/to/tracker$urlskip=?url`

Apply the `urlskip=` redirect whether the network request is blocked or not:
`||example.com/path/to/tracker$urlskip==blocked ?url`

This is potentially a breaking change, but I think it's for the best. For instance, I received a scam email with a link to `googleadservices.com`, which is a legitimate (i.e non-malicious) domain name. Having `urlskip=` for such domain name makes sense, however we wouldn't want to automatically redirect to scam sites through what appears as a legitimate server. With the above proposal, the redirect wouldn't occur by default, only when the navigation to the legitimate domain name is not blocked. The new step `=blocked` would indicate that the navigation should occur for blocked request as well.

**Comment by stephenhawk8054** [comment]
Yeah I was just careful to minimize the risk of FP matches since it's a type of filters that auto load a different link to users.

**Comment by garry-ut99** [comment]
> So for a filter like `||com/*/http$doc`, the extracted token can be either `com` or `http` (both are considered "bad" token by the way given there occurrences).

`||com/*/https$doc` works (if encoded URL has `http` replaced with `https` like in my previous comment), so `http` is a bad token, while `https` seems to be a good token somehow...

**Comment by stephenhawk8054** [comment]
> With the above proposal, the redirect wouldn't occur by default, only when the navigation to the legitimate domain name is not blocked. The new step =blocked would indicate that the navigation should occur for blocked request as well.

That looks better actually. I'm also wondering myself how to distinguish the malicious vs trackers ones, but the line is really blurred and tough.

Most of the `urlskip` filters are based on Brave's debounce database: https://github.com/brave/adblock-lists/blob/master/brave-lists/debounce.json with the check again on how those tracking domains are used in real sites.

The thing is once the `urlskip` is used, we can't acknowledge or prevent how those domains could be used later.

**Comment by stephenhawk8054** [comment]
A safer approach might be displaying in the strict-blocked page the decoded/destination link for blocked domains and warn users to check again the destination link before continuing.

It's just strict-blocked page again, but with the help of `urlskip` to assert the destination link to users.

This is the safest approach I can think of, since I still feel quite risky with `urlskip`.

**Comment by gorhill** [comment]
> `||com/*/https$doc works`

It works because regarding token "badness", `https` > `com` > `http`. You can see the token "badness" score [here](https://github.com/gorhill/uBlock/blob/1.60.0/src/js/static-net-filtering.js#L3308).

**Comment by Yuki2718** [comment]
badtoken will still be used if no better is available.

**Comment by gorhill** [comment]
> It's just strict-blocked page again, but with the help of `urlskip` to affirm the destination link to users.

Yes, this will requires more work which I can't undertake immediately, but I will go ahead and change the behavior of `urlskip=` to apply only when a network requests would go through anyways, in which case it doesn't add risks. The new directive `=blocked` will be added for cases where we want to redirect blocked network requests.

**Comment by stephenhawk8054** [comment]
> apply only when a network requests would go through anyways

So, in case of blocked network requests, if users continue to click on "Proceed", would `urlskip` (without `=blocked`) be able to activate on that network request?

**Comment by gorhill** [comment]
> if users continue to click on "Proceed", would `urlskip` (without `=blocked`) be able to activate on that network request?

Yes, matching `urlskip=` will apply on proceed-ed pages as they are no longer blocked. I've renamed the directive `-blocked`, as in "remove the block status of the network request".

**Comment by gorhill** [comment]
> A safer approach might be displaying in the strict-blocked page the decoded/destination link in case of already-blocked domains and warn users to check again the destination link before continuing.

I looked into how much work this would require, and I think I feel comfortable to include this before the next stable release, which I would like it to happen after a week from now.

I need help with the wording and how to best present this to end user. This is a quick draft:

![image](https://github.com/user-attachments/assets/8d84fb69-4736-4be5-940e-dcb6098632fe)


Filters used:

    ||example.com^
    ||example.com/path/to/tracker$urlskip=?url

**Comment by stephenhawk8054** [comment]
Yeah, I'm not fluent at English wording. Maybe "uBO will navigate directly to the following page. Please check the link again before you proceed:" ?

**Comment by gorhill** [comment]
> Please check the link again before you proceed

This sounds as if we are suggesting to click proceed. Now I am considering:

> The blocked page wants to redirect to another site. If you choose to proceed, you will navigate directly to:
> {{url}}

I am also wondering if in case of matching `urlskip` we should not show the section "Because of the following filter...".

**Comment by garry-ut99** [comment]
> https://github.com/gorhill/uBlock/commit/d04dc4c767721fb13d91a67cb62ecad9b9219103

Doesn't work, the redirection still occurs even without `-blocked` directive.

**Comment by stephenhawk8054** [comment]
> I am also wondering if in case of matching `urlskip` we should not show the section "Because of the following filter...".

I think we should keep that. The page is blocked mainly because of that filter, so giving that exact info to users is still better. The `urlskip` is just an assistant for users. And if somehow the `urlskip` matches some blockings from "badware" list, it will also help raise more alerts to users and easier for them to feedback.

**Comment by gorhill** [comment]
> Doesn't work

Looks like I made a last second change without staging the modified file, there should be a `, true` argument at [that line](https://github.com/gorhill/uBlock/blob/1.60.1rc0/src/js/pagestore.js#L959).

**Comment by Yuki2718** [comment]
It's a bit unclear what will happen if user checked "Don't warn me again about this site" in this case.

**Comment by garry-ut99** [comment]
Works now. Blocking works for all request types, however `-blocked` works only with `doc` type requests, but not for `media` or `other` types, for example:

After adding:
```adb
||dts.podtrac.com/redirect.mp3/hozana.si/mp3
||podtrac.com^*redirect.mp3/$urlskip=/\/redirect\.mp3\/(.*?\.mp3\b)/ +https,badfilter
||podtrac.com^*redirect.mp3/$urlskip=-blocked /\/redirect\.mp3\/(.*?\.mp3\b)/ +https
```

and then on `https://hozana.si/` (as per https://github.com/uBlockOrigin/uBlock-issues/issues/3206#issuecomment-2408604700) after: 
- 1] Pressing play button (`media`)
- 2] Saving sound as file (from context menu) (`other`)

the redirection still doesn't occur for `https://dts.podtrac.com/redirect.mp3/hozana.si/mp3/berila/794.mp3` and it's still blocked despite `-blocked` directive was used. 

The description of the commit clearly mentions "network requests" without limiting it to `doc` type, so it seems it's a bug, unless `-blocked` was supposed to work only with `doc` type - in such a case commit description was wrongly / mistakenly formulated.

**Comment by gorhill** [comment]
> and then on `https://hozana.si/`

Which exact URL do I need to navigate to?

**Comment by garry-ut99** [comment]
> > and then on `https://hozana.si/`
> 
> Which exact URL do I need to navigate to?

This is already the exact URL, there are embed audio players on the right side of page.

**Comment by gorhill** [comment]
> already the exact URL

Sorry, didn't realize this.

> the redirection still doesn't occur for `https://dts.podtrac.com/redirect.mp3/hozana.si/mp3/berila/794.mp3` and it's still blocked despite `-blocked` directive was used.

Now I am questioning that we really need `-blocked`. The main use was to not block certain strict-blocked documents, but since now users are provided the information about the incoming redirects, I don't see the need anymore about this directive.

Should a case like so arise:

    ||dts.podtrac.com/redirect.mp3/hozana.si/mp3
    ||podtrac.com^*redirect.mp3/$urlskip=-blocked /\/redirect\.mp3\/(.*?\.mp3\b)/ +https

Then the this can already be taken care of without `-blocked`:

    ||dts.podtrac.com/redirect.mp3/hozana.si/mp3
    ||podtrac.com^*redirect.mp3/$urlskip=/\/redirect\.mp3\/(.*?\.mp3\b)/ +https
    ||dts.podtrac.com/redirect.mp3/hozana.si/mp3$badfilter

**Comment by garry-ut99** [comment]
Works fine now (with both `media` and `other`, btw. even redirecting works with `other` now).

<hr>

> `||dts.podtrac.com/redirect.mp3/hozana.si/mp3$badfilter`

Or `@@||dts.podtrac.com/redirect.mp3/hozana.si/mp3` (if not blocked by `important`).

> An argument to keep `-blocked` is that the request will stay blocked should there be no matching `urlskip=` filter.

Yes, it would be good to find some real world cases which make usage of `-blocked`.  
Maybe we should wait some time to see whether filter lists devs will encounter such cases or not. Who knows.
Btw. it's kinda similiar mechanism to how `redirect` & `redirect-rule` work, but in the opposite direction.

**Comment by stephenhawk8054** [comment]
@gorhill In this link

```yml
http://go.redirectingat.com/?id=355X561&test=off&xcust=94d11aa0976a11efb985b213d4e1265c0INT&url=https%3A%2F%2Fwww.menards.com%2Fmain%2Fstorage-organization%2Fgarage-outdoor-storage%2Fbike-storage%2Ftailgate-bike-rack-carrier-protection-pad-5-bike%2Frmbp500%2Fp-1642874308595068-c-12651.htm
```

Should we display the full link of redirected link in case it's too long?

![image](https://github.com/user-attachments/assets/7f580975-8ab9-45d3-9352-ea917b0fb62f)

**Comment by garry-ut99** [comment]
By the way, another thing is that there seem to be an inconsistency there: the final link is not embeded in tag code, unlike the initial url and the blocking filter, curious whether this inconsitency is intentional or the code tag has been forgotten to be added.

**Comment by gorhill** [comment]
On mobile, not preventing multiline pushes the Back/Proceed button below the viewport and I think this would be an issue for a lot of people who would think they have to click the link as the only choice.

The link is not in `code` tags because it's likely to be safe since we added a filter specifically for this. It's the same for links revealed by the magnifier.

**Comment by gorhill** [comment]
I can let the link wrap but I will limit the height on mobile.

**Comment by gorhill** [comment]
> why ever the landing link have to be clickable

The page is telling me it's going to navigate there, as a user I would be wondering why not to just make the link clickable then.

> easy copyable for whatever needs

Right-click, "Copy link" / "Copy link address"

**Comment by garry-ut99** [comment]
> The page is telling me it's going to navigate there, as a user I would be wondering why not just make the link clickable then.

Because like I already mentioned, there already exists clickable _Continue_ button for the same purpose...
But it's fine, as there exist other reasons why to make the link copyable, which I mentioned and which you confirmed, and I take your reply as the confirmation.

**Comment by stephenhawk8054** [comment]
@gorhill I see the domains are marked bold in the strict-blocked page, is it intended?

**Comment by injust** [comment]
I saw that https://github.com/gorhill/ublock/wiki/Static-filter-syntax#urlskip has this example:

```adblock
||example.com/path/to/tracker$urlskip=-blocked ?url
```

Is there any specific reason for `-blocked` to be placed before the `?url` directive in the example? It still works if placed after `?url`.

---

This got me thinking: If you specify multiple `?name` directives, how are the `-base64`, `-uricomponent`, and `-blocked` directives applied?

Consider `?foo ?bar -base64`. What will this do?

1. Extract the value of `foo`, then extract the value of `bar` and base64 decode it
2. Extract the value of `foo` and base64 decode it, then extract the value of `bar` and base64 decode it

I haven't tried this and haven't run into any real-life examples, but I wonder if the order of the directives should be formalized a bit.

**Comment by stephenhawk8054** [comment]
@gorhill Sorry, is it possible to automatically use `https` for queries (`?` and `&`) cases? Because I don't really know when a redirection domain uses `http` only instead. For example in this case: https://github.com/uBlockOrigin/uAssets/commit/d8457a1af2919cfb5e5fe3332ce1a04ccfa63802 I need to add `+https` to make it work.

**Comment by gorhill** [comment]
You mean if the protocol is present and is `http:`, to automatically upgrade to `https:`?

**Comment by stephenhawk8054** [comment]
Yeah exactly.

**Comment by stephenhawk8054** [comment]
@gorhill In this link:

```yml
https://theatre2.prospect2.com/Prod/link-tracker?redirectUrl=aHR0cCUzQSUyRiUyRnRoZWF0cmUyLm9yZyUyRnN1YnNjcmliZSUzRnV0bV9zb3VyY2UlM0RBY3RpdmVDYW1wYWlnbiUyNnV0bV9tZWRpdW0lM0RlbWFpbCUyNnV0bV9jb250ZW50JTNEVGhlYXRyZVNxdWFyZWQlMkJwcm9kdWN0aW9uJTJCb2YlMkJUaWdlciUyQlN0eWxlJTI1MjElMkJiZWdpbnMlMkJGZWJydWFyeSUyQjIzJTI2dXRtX2NhbXBhaWduJTNEUFIlMjUzQSUyQlRpZ2VyJTJCU3R5bGU=&sig=8MyAZqfWi4D4y6jQ74HofpnnUiWyDBP7zh1Mhb1PDywm&iat=1644533531&a=%7C%7C1001253958%7C%7C&account=theatre2%2Eactivehosted%2Ecom&email=x5iTFbWD27OamnRX5Z9MOqSyIXu03Akc1Z8hSpIr818%3D&s=3949e035518f7fd77b4f62360256cda3&i=368A382A2A4505
```

The URL inside `redirectUrl` parameter after base64-decode is

```yml
http%3A%2F%2Ftheatre2.org%2Fsubscribe%3Futm_source%3DActiveCampaign%26utm_medium%3Demail%26utm_content%3DTheatreSquared%2Bproduction%2Bof%2BTiger%2BStyle%2521%2Bbegins%2BFebruary%2B23%26utm_campaign%3DPR%253A%2BTiger%2BStyle
```

which needs another layer of uri-decode. Could there be any ways to address this?

Reference: https://github.com/brave/adblock-lists/issues/993

**Comment by stephenhawk8054** [comment]
> `-uricomponent` doesn't work?

I don't think so. I tested

```adb
/Prod/link-tracker?redirectUrl=$doc,urlskip=?redirectUrl -base64 -uricomponent
```

but it doesn't work.

**Comment by stephenhawk8054** [comment]
Oh wait, sorry my bad. The destination link is `http`, not `https`, so this works

```
/Prod/link-tracker?redirectUrl=$doc,urlskip=?redirectUrl -base64 -uricomponent
```

**Comment by stephenhawk8054** [comment]
@gorhill Sorry, currently in the result links of Bing search, the URLs look like this:

```ini
https://www.bing.com/ck/a?!&&p=ef7ef125927f4fd6e7fd462510d6c90eccceb80d05a10a1250756826aa14611dJmltdHM9MTczMTk3NDQwMA&ptn=3&ver=2&hsh=4&fclid=26de3a35-cd30-618a-182b-2f08cce2603e&psq=lenovo+thinkpad+z16&u=a1aHR0cHM6Ly93d3cubGVub3ZvLmNvbS9qcC9qYS9wL2xhcHRvcHMvdGhpbmtwYWQvdGhpbmtwYWQtei1zZXJpZXMvdGhpbmtwYWQtejE2LSgxNi1pbmNoLWFtZCkvbGVuMTAxdDAwMzc_bXNvY2tpZD0yNmRlM2EzNWNkMzA2MThhMTgyYjJmMDhjY2UyNjAzZQ&ntb=1
```

I tried

```adb
||bing.com/ck/a*^u=a1aHR0c$doc,urlskip=/[?&]u=a1(aHR0c[^\/&#]+)/ -base64
```

but not succeeded. Can you reproduce?

**Comment by gwarser** [comment]
@stephenhawk8054 there is an error or some special encoding in base64 string, around the `_`.

**Comment by stephenhawk8054** [comment]
Ah you're right, I forgot to check the `atob`, it indeed shows error. Not sure what `https://www.base64decode.org/` uses since it doesn't show error.

**Comment by gwarser** [comment]
`_` -> `/`

**Comment by freezer2022** [comment]
Information:  a related subthread: #3499 with a discussion related to added support for a step to extract hash (`#`) part of a URL. (by the way, there was no need to create a separate thread for it, as it just causes unnecessary fragmentation of the discussion).

**Comment by stephenhawk8054** [comment]
> Sorry, is it possible to automatically use `https` for queries (`?` and `&`) cases? Because I don't really know when a redirection domain uses `http` only instead. For example in this case: [uBlockOrigin/uAssets@d8457a1](https://github.com/uBlockOrigin/uAssets/commit/d8457a1af2919cfb5e5fe3332ce1a04ccfa63802) I need to add `+https` to make it work.

@gorhill Oh sorry, looks like this could cause issues on chromium for sites that don't have HTTPS: https://www.reddit.com/r/uBlockOrigin/comments/1j6obb4/bing_search_for_kevin_eats_results_in_secure/

I don't know if there's any solution for this. The number of these HTPP-only sites that appears on redirection links are quite small I think.

**Comment by gorhill** [comment]
Two possible solutions when `http:` in final URL is detected:
- I remove the code enforcing `https:`; or
- I abort transformation and return original URL

**Comment by stephenhawk8054** [comment]
How about some ways that we can set to not upgrade to HTTPS based on the destination domain?

 For example

```adb
||bing.com/ck/a*^u=a1ahr0c$document,urlskip=/[?&]u=a1(aHR0c[^&#]+)/ -safebase64 -https:kevineats.com
```

**Comment by stephenhawk8054** [comment]
@gorhill In that case, can we make `urlskip` work with `http` link? 

If I understand correctly from `if ( urlfinal.protocol !== 'http:' ) { return; }`, if the destination link is `http`, the whole `urlskip` won't work so users still both get tracked and destination link is `http` link anyway?

**Comment by gorhill** [comment]
It will work if the final URL is either `https:` or `http:`, it won't wotk if the final URL is anything else, e.g. `javascript:`.

**Comment by injust** [comment]
Sorry for the noise, I posted a comment and then deleted it. I didn't realize that query params get implicitly decoded by the browser, not uBlock.
